carbonplan_tonyear/core.py

```diff
@@ -4,40 +4,69 @@
 import numpy as np
 
 
-def get_baseline_curve(curve_name):
+def get_baseline_curve(curve_name: str, t_horizon: int = 1001) -> np.ndarray:
+    """Build the baseline curve
+
+    Parameters
+    ----------
+    curve_name : str
+        Name of baseline curve
+    t_horizon : int
+        Length of the time horizon (years)
+
+    Returns
+    -------
+    baseline_curve : np.ndarray
+        Baseline curve in the form of an 1D array
+    """
+
+    if t_horizon <= 0:
+        raise ValueError('t_horizon must be a postive integer')
+
     if curve_name == 'joos_2013':
         # parameters from Joos et al., 2013 (Table 5)
         # https://doi.org/10.5194/acp-13-2793-2013
         a = [0.2173, 0.2240, 0.2824, 0.2763]
         tau = [0, 394.4, 36.54, 4.304]
-        t_horizon = np.arange(1001)
     elif curve_name == 'ipcc_2007':
         # parameters from IPCC AR4 2007 (Chapter 2, page 213)
         # https://www.ipcc.ch/site/assets/uploads/2018/02/ar4-wg1-chapter2-1.pdf
         a = [0.217, 0.259, 0.338, 0.186]
         tau = [0, 172.9, 18.51, 1.186]
-        t_horizon = np.arange(1001)
     elif curve_name == 'ipcc_2000':
         # parameters from IPCC LULUCF Special Report 2000 (Chapter 2.3.6.3, Footnote 4)
         # https://archive.ipcc.ch/ipccreports/sres/land_use/index.php?idp=74
         a = [0.175602, 0.137467, 0.18576, 0.242302, 0.258868]
         tau = [0, 421.093, 70.5965, 21.42165, 3.41537]
-        t_horizon = np.arange(1001)
     else:
-        raise ValueError('No baseline curve parameters by the name \'' + curve_name + '\'.')
+        raise ValueError(f'No baseline curve parameters by the name {curve_name}.')
 
-    baseline_curve = [a[0]] * len(t_horizon)
-    for t in t_horizon:
+    baseline_curve = np.full(t_horizon, a[0])
+    for t in range(t_horizon):
         for i in np.arange(1, len(a)):
-            baseline_curve[t] = baseline_curve[t] + (a[i] * np.exp(-t / tau[i]))
+            baseline_curve[t] += a[i] * np.exp(-t / tau[i])
     return baseline_curve
 
 
-def get_discounted_curve(discount_rate, curve):
-    return [x / math.pow(1 + discount_rate, i) for i, x in enumerate(curve)]
+def get_discounted_curve(discount_rate: float, curve: np.ndarray) -> np.ndarray:
+    """Get discounted curve
+
+    Parameters
+    ----------
+    discount_rate : float
+        Discount rate expressed as a fraction.
+    curve : np.ndarray
+
+    Returns
+    -------
+    discounted_curve : np.ndarray
+        Curve with discount rate applied.
+    """
+    return np.asarray([x / math.pow(1 + discount_rate, i) for i, x in enumerate(curve)])
```

Comment:
we could use `np.power` instead of `math.pow` if we wanted to get rid of the import. it might even be possible to vectorize this then? 

```python
return np.power(1 + discount_rate, np.arange(len(curve))
```

Reply:
I was thinking the same thing but was trying to be less disruptive. But wouldn't it be: 

```python
curve / np.power(1 + discount_rate, np.arange(len(curve)))
```

Reply:
oof. of course.